### PR TITLE
Merge r1.11 main

### PR DESCRIPTION
**Pull request by ericharper**
# What does this PR do ?

Add a one line overview of what this PR aims to accomplish.

**Collection**: [Note which collection this PR will affect]

# Changelog 
- Add specific line by line info of high level changes in this PR.

# Usage
* You can potentially add a usage example below

```python
# Add a code snippet demonstrating how to use this 
```

# Before your PR is "Ready for review"
**Pre checks**:
- [ ] Make sure you read and followed [Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md)
- [ ] Did you write any new necessary tests?
- [ ] Did you add or update any necessary documentation?
- [ ] Does the PR affect components that are optional to install? (Ex: Numba, Pynini, Apex etc)
  - [ ] Reviewer: Does the PR have correct import guards for all optional libraries?
  
**PR Type**:
- [ ] New Feature
- [ ] Bugfix
- [ ] Documentation

If you haven't finished some of the above items you can still open "Draft" PR.

## Who can review?

Anyone in the community is free to review the PR once the checks have passed. 
[Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md) contains specific people who can review PRs to various areas.

# Additional Information
* Related to # (issue)

### CODE REVIEW

File: tutorials/tts/FastPitch_GermanTTS_Training.ipynb [hunk 1]
```diff
@@ -543,7 +543,7 @@
     "    punct_post_process: true\n",
     "\n",
     "  text_tokenizer:\n",
-    "    _target_: nemo.collections.common.tokenizers.text_to_speech.tts_tokenizers.GermanCharsTokenizer\n",
+    "    _target_: nemo.collections.tts.torch.tts_tokenizers.GermanCharsTokenizer\n",
```
Comment (XuesongYang) [inline tutorials/tts/FastPitch_GermanTTS_Training.ipynb:546]: keep `main`.

File: scripts/dataset_processing/tts/hui_acg/ds_conf/ds_for_fastpitch_align.yaml [hunk 2]
```diff
@@ -38,7 +38,7 @@ dataset:
     punct_post_process: true
 
   text_tokenizer:
-    _target_: nemo.collections.common.tokenizers.text_to_speech.tts_tokenizers.GermanCharsTokenizer
+    _target_: nemo.collections.tts.torch.tts_tokenizers.GermanPhonemesTokenizer
```
Comment (XuesongYang) [inline scripts/dataset_processing/tts/hui_acg/ds_conf/ds_for_fastpitch_align.yaml:41]: keep `main`.

File: nemo/collections/tts/torch/g2ps.py [hunk 3]
```diff
@@ -12,6 +12,452 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-# TODO (xueyang): deprecate this file since no other places import modules from here anymore. However,
-#  all checkpoints uploaded in ngc used this path. So it requires to update all ngc checkpoints g2p path as well.
-from nemo_text_processing.g2p.modules import IPAG2P, BaseG2p, EnglishG2p
```
Comment (redoctopus) [inline nemo/collections/tts/torch/g2ps.py:17]: `main` should be kept here, the G2P modules have moved to a more generic location

File: nemo/collections/tts/torch/tts_dataset.yaml [hunk 4]
```diff
@@ -41,6 +41,6 @@ tts_dataset:
     add_blank_at: null
     pad_with_space: True
     g2p:
-      _target_: nemo_text_processing.g2p.modules.EnglishG2p
```
Comment (redoctopus) [inline nemo/collections/tts/torch/tts_dataset.yaml:44]: `main` has the correct new path

File: nemo/collections/tts/models/fastpitch.py [hunk 5]
```diff
@@ -46,7 +46,7 @@
 
 @dataclass
 class G2PConfig:
-    _target_: str = "nemo_text_processing.g2p.modules.EnglishG2p"
```
Comment (redoctopus) [inline nemo/collections/tts/models/fastpitch.py:49]: `main` has the correct new path

File: nemo/collections/tts/torch/tts_tokenizers.py [hunk 6]
```diff
@@ -12,13 +12,587 @@
...
-# TODO (xueyang): deprecate this file since no other places import modules from here anymore. However,
-#  all checkpoints uploaded in ngc used this path. So it requires to update all ngc checkpoints path as well.
-from nemo.collections.common.tokenizers.text_to_speech.tts_tokenizers import (
-    BaseCharsTokenizer,
-    BaseTokenizer,
-    EnglishCharsTokenizer,
-    EnglishPhonemesTokenizer,
-    GermanCharsTokenizer,
-    IPATokenizer,
-)
```
Comment (redoctopus) [inline nemo/collections/tts/torch/tts_tokenizers.py:24]: Please keep `main`, these modules have been moved

File: tutorials/tts/FastPitch_MixerTTS_Training.ipynb [hunk 7]
```diff
@@ -1,612 +1,611 @@
...
+        },
+        {
+            "cell_type": "code",
+            "execution_count": null,
+            "id": "7ef87e31",
+            "metadata": {},
+            "outputs": [],
+            "source": [
+                "from nemo.collections.tts.torch.g2ps import EnglishG2p\n",
+                "from nemo.collections.tts.torch.data import TTSDataset\n",
```
Comment (redoctopus) [inline tutorials/tts/FastPitch_MixerTTS_Training.ipynb:202]: `main` has the correct updated paths.

File: scripts/tts_dataset_files/ipa_cmudict-0.7b_nv22.08.txt [hunk 8]
```diff
@@ -57525,8 +57525,6 @@ HURRAY  həˈɹeɪ
 HURRELL  ˈhɔɹəl
 HURRI  ˈhɝi
 HURRICANE  ˈhɝəˌkeɪn
-HURRICANE(1)  ˈhʌɹəˌkeɪn
-HURRICANE'S  ˈhɝəˌkeɪnz
```
Comment (redoctopus) [inline scripts/tts_dataset_files/ipa_cmudict-0.7b_nv22.08.txt:57529]: @ekmb can you double-check which version here is more up-to-date?

**Review by redoctopus — COMMENTED** [review]
Have a few things to keep from main since they were most recently updated there.

File: nemo/collections/tts/torch/g2ps.py (hunk 3 above)
Comment (redoctopus) [reply nemo/collections/tts/torch/g2ps.py:17]: @AlexGrinch could you go through this merge and update the branch based on your changes from PR #4690?

**Comment by ericharper** [comment]
Closed in favor of #4920

File: examples/nlp/language_modeling/conf/megatron_t5_prompt_learning.yaml [hunk 9]
```diff
@@ -12,6 +12,7 @@ trainer:
   max_steps: -1
   log_every_n_steps: 10
   val_check_interval: 1.0
+  accumulate_grad_batches: 1
```
Comment (XuesongYang) [inline examples/nlp/language_modeling/conf/megatron_t5_prompt_learning.yaml:15]: I am comparing my branch (https://github.com/NVIDIA/NeMo/pull/4920) to yours. From my research, this config should be removed. Could you please confirm?

Comment (ericharper) [reply examples/nlp/language_modeling/conf/megatron_t5_prompt_learning.yaml:15]: It can be removed. It's not hurting anything having it there though.

Comment (XuesongYang) [reply examples/nlp/language_modeling/conf/megatron_t5_prompt_learning.yaml:15]: ok. i will keep what `main` did then.

**Pull request closed without merging**